### ISSUE TITLE
Stop (mis)using pistachio_bub for all pistachio based boards

### ISSUE THREAD

**Issue by Ham22** [FlowM2M/u-boot#22]
Pistachio was never added as a SoC and instead we added everything in the pistachio_bub board setup. This has meant as new boards have been added they have been hacked in based on the pistachio_bub config rather than based on the pistachio SoC.

It would be nice if we could spend some time fixing this once and for all otherwise the problem is just going to get worse.

**Comment by abhimanyuv1** [comment]
I don't think we should remove pistachio_bub completely as later we might add new board and may need new config.  may be better to rename to more generic pistachio_common.h or just pistachio.h

**Comment by Ham22** [comment]
The aim isn't to remove pistachio bub. The issue is that u-boot is designed to let you add SoCs and then boards based on those SoCs so you have something nice like this:
```
+------------------------------+
|  marduk  |  beetle  |  bub   |
|------------------------------|
|           pistahcio          |
+------------------------------+
```

but instead we have no SoC definition and just a single board of pistachio bub which has had some other board support hacked into it by passing through different config options:
```
+--------------------------------------------------+
|  pistachio bub (with hacks for marduk and beetle |
+--------------------------------------------------+
```
